store/url.go

```diff
@@ -0,0 +1,49 @@
+package store
+
+import (
+	"crypto/tls"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/argoproj/argo-events/pkg/apis/sensor/v1alpha1"
+)
+
+// URLReader implements the ArtifactReader interface for urls
+type URLReader struct {
+	urlArtifact *v1alpha1.URLArtifact
+}
+
+// NewURLReader creates a new ArtifactReader for workflows at URL endpoints.
+func NewURLReader(urlArtifact *v1alpha1.URLArtifact) (ArtifactReader, error) {
+	if urlArtifact == nil {
+		panic("URLArtifact cannot be empty")
+	}
+	return &URLReader{urlArtifact}, nil
+}
+
+func (reader *URLReader) Read() ([]byte, error) {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
```

Comment:
Do you think we should be able to support providing options/configuration to the `http.Client` through the `UrlArtifact`?